core/contracts/financial_templates/PricelessPositionManager.sol

```diff
@@ -197,11 +226,114 @@ contract PricelessPositionManager is Testable {
         tokenCurrency.burn(numTokens.rawValue);
     }
 
+    /**
+    * @notice After expiration of the contract the DVM is asked what for the prevailing price at the time of
+    * expiration.
+    */
+    function expire() public onlyPostExpiration() {
+        _requestOraclePrice(expirationTimestamp);
+    }
+
+    /**
+     * @notice After a contract has passed maturity all token holders can redeem their tokens for underlying at
+     * the prevailing price defined by the DVM from the `expire` function. This Burns all tokens from the caller
+     * of `tokenCurrency` and sends back the proportional amount of `collateralCurrency`.
+     */
+    function settleExpired() public onlyPostExpiration() {
```

Comment:
Does this method cover the case where there's not enough collateral in the contract to cover all the token holders?

Reply:
not as of yet.